### PR TITLE
Updates for testing towncrier 21.3

### PR DESCRIPTION
**Pull request by Cadair**
This tests the new sphinx-changelog 1.1.0rc1 which uses towncrier 21.3

Close #11728

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -767,97 +767,12 @@ Other Changes and Additions
 
 - Removed dependency on scikit-image. [#10214]
 
-
-4.0.6 (unreleased)
```
Comment (Cadair) [inline CHANGES.rst:771]: I didn't think we should have unreleased sections in the changelog now we are using towncrier, but can revert if I am missing something.

**Comment by Cadair** [comment]
Seems to work: https://astropy--11726.org.readthedocs.build/en/11726/changelog.html

**Comment by Cadair** [comment]
I have just pushed the final 1.1.0 release.

File: setup.cfg [hunk 2]
```diff
@@ -88,7 +88,7 @@ docs =
     PyYAML>=3.13
     scipy>=1.1
     matplotlib>=3.1,!=3.4.0
-    sphinx-changelog
+    sphinx-changelog>=1.1.0rc1
```
Comment (pllim) [inline setup.cfg:91]: Do we still need to pin to RC? Can we pin to stable yet?

Comment (Cadair) [reply setup.cfg:91]: Well this is the minimum and it's technically correct, tho we can bump to 1.1.0.

```suggestion
    sphinx-changelog>=1.1.0
```

**Review by pllim — APPROVED** [review]
I think it looks fine at https://astropy--11726.org.readthedocs.build/en/11726/changelog.html . But let's see what others say.

**Comment by pllim** [comment]
Also I am not sure about the milestone...

**Comment by Cadair** [comment]
We should backport this to all supported branches imo

**Review by saimn — APPROVED** [review]
Sounds good to me :+1:

**Comment by eteq** [comment]
backported to v4.3.x in 3977cea6cf8a8337522a4d138389a35a1129cac3

**Comment by astrofrog** [comment]
For the record the backport of this to LTS was incomplete (2c0f690993119d97248ec9c17d6626dd94cfab5b) - I've added the missing changes to https://github.com/astropy/astropy/pull/12075

**Pull request merged**